lib/logreader.c

```diff
@@ -196,7 +196,7 @@ log_reader_io_handle_in(gpointer s)
 {
   LogReader *self = (LogReader *) s;
 
-  log_reader_stop_watches(self);
+  poll_events_suspend_watches(self->poll_events);
```

Comment:
Fd polling will be restored in `update_watches`, 
- in `log_reader_start_watches_if_stopped` which no longer will be called unless `log_reader_stop_watches` is not called prior.
I think a registered immediate_check (`restart_task`) must be stopped if we got a poll event.
- in `poll_events_update_watches` fd handlers changed according to `log_proto_server_prepare` result

Shouldn't be the `log_reader_stop_watches` changed to use `poll_events_suspend` instead of `poll_events_stop_watches`?
Of course in deinit, we need to call unregister, so the same stop function wouldn't be feasible.

Reply:
We've addressed this while we've worked on the refactor.